<launch>
  <!--Publish a pcd as pointcloud and run clustering node, must provide tf-->
  <arg name="name" default="takeshiBagPack1"/>

  <node name="segmenter" pkg="space_quantization" type="segmentation_node" args="" output="screen">
    <remap from="cloud" to="/hsrb/cloud"/>
    <param name="nClusters" value="16" type="int"/>
    <param name="iterations" value="10" type="int"/>
    <param name="visualizeCentroids" value="True" type="bool"/>
    <param name="visualizeSegCloud" value="false" type="bool"/>
    <param name="freeThr" value="0.05" type="double"/>
  </node>

  <!-- <node pkg="tf" type="static_transform_publisher" name="link_broadcaster_table" args="-0.005 1.115 0.802 -0.037 0.821 -0.569 -0.015 base_link head_rgbd_sensor_rgb_frame 100"/> -->

  <!-- <node pkg="tf" type="static_transform_publisher" name="link_broadcaster_backpack" args="0.016 -0.073 0.979 -0.009, 0.887, -0.462, -0.004 base_link head_rgbd_sensor_rgb_frame 100"/> -->

  <node pkg="tf" type="static_transform_publisher" name="link_broadcaster_byron" args="0.020, -0.010, 1.326, -0.139, 0.810, -0.563, 0.086 base_link head_rgbd_sensor_rgb_frame  100"/>
<!-- head_rgbd_sensor_rgb_frame -->
  <node name="cloudPublisher" pkg="pcl_ros" type="pcd_to_pointcloud" args="$(find space_quantization)/cloud/$(arg name).pcd " output="screen">
    <remap from="cloud_pcd" to="/hsrb/cloud"/>
    <param name="frame_id" value="head_rgbd_sensor_rgb_frame" type="str"/>
  </node>

</launch>
